client/components/AddTestToQueueWithConfirmation/index.jsx

```diff
@@ -31,17 +53,86 @@ function AddTestToQueueWithConfirmation({
         </>
     );
 
+    const closeWithUpdate = async () => {
+        setShowConfirmation(false);
+        await triggerUpdate();
+        setTimeout(() => {
+            if (buttonRef?.current) {
+                buttonRef.current.focus();
+            }
+        }, 0);
+    };
+
+    const renderConfirmation = () => {
+        const actions = [];
+        if (hasAutomationSupport) {
+            actions.push(
+                {
+                    label: 'Add and run later',
+                    onClick: async () => {
+                        await addTestToQueue();
+                        await closeWithUpdate();
+                    }
+                },
+                {
+                    label: 'Add and run with bot',
+                    onClick: async () => {
+                        const testPlanReport = await addTestToQueue();
+                        await scheduleCollectionJob(testPlanReport);
+                        await closeWithUpdate();
+                    }
+                }
+            );
+        }
+        return (
+            <BasicModal
+                dialogClassName={'add-test-to-queue-confirmation'}
+                show={showConfirmation}
+                title={feedbackModalTitle}
+                content={feedbackModalContent}
+                closeLabel={hasAutomationSupport ? 'Cancel' : 'Ok'}
+                staticBackdrop={true}
+                actions={actions}
+                useOnHide
+                handleClose={async () => {
+                    if (hasAutomationSupport) {
+                        setShowConfirmation(false);
+                    } else {
+                        await closeWithUpdate();
+                    }
+                }}
+            />
+        );
+    };
+
     const addTestToQueue = async () => {
+        let tpr;
         await triggerLoad(async () => {
-            await addTestPlanReport({
+            const res = await addTestPlanReport({
                 variables: {
                     testPlanVersionId: testPlanVersion.id,
                     atId: at.id,
                     browserId: browser.id
                 }
             });
+            const testPlanReport =
+                res?.data?.findOrCreateTestPlanReport?.populatedData
+                    ?.testPlanReport ?? null;
+            tpr = testPlanReport;
```

Comment:
Maybe another piece of tech debt: update `useTriggerLoad`'s callback-accepting function to forward the callback's return value. (Also test `useTriggerLoad` while we're at it.)

Reply:
I agree with this. This code exists in the app in a different location prior to this PR so this might not necessarily be automation tech debt but general app tech debt. Not sure if this distinction is meaningful.